res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -2134,16 +2360,56 @@ PioneerDDJSX.shiftFxBeatsKnob = function(channel, control, value, status, group)
     newValue = actValue + delta;
 
     if (newValue >= 0.00 && newValue <= 1.00) {
-        engine.setValue("[EffectRack1_EffectUnit" + (unit + 1) + "]", "mix", newValue);
+        engine.setParameter("[EffectRack1_EffectUnit" + (unit + 1) + "]", "mix", newValue);
     }
 };
 
-PioneerDDJSX.softTakeoverEmulation = function(unit, index, currentValue) {
-    var deltaToActive = currentValue - PioneerDDJSX.fxParamsActiveValues[unit][index];
 
-    if (Math.abs(deltaToActive) < 15) {
-        PioneerDDJSX.fxParamsActiveValues[unit][index] = currentValue;
-        return true;
+///////////////////////////////////////////////////////////////
+//                          SLICER                           //
+///////////////////////////////////////////////////////////////
+
+PioneerDDJSX.slicerBeatActive = function(value, group, control) {
+    var deck = PioneerDDJSX.channelGroups[group],
+        bpm = engine.getValue(group, "bpm"),
+        playposition = engine.getValue(group, "playposition"),
+        duration = engine.getValue(group, "duration"),
+        slicerPos = 0,
+        ledBeatState = true;
+
+    if (engine.getValue(group, "beat_closest") === engine.getValue(group, "beat_next")) {
+        return;
+    }
+
+    PioneerDDJSX.slicerBeatsPassed[deck] = Math.round((playposition * duration) * (bpm / 60));
+    slicerPos = PioneerDDJSX.slicerBeatsPassed[deck] % 8;
+
+    if (PioneerDDJSX.padMode[deck] === 2) {
+        if (PioneerDDJSX.slicerType[deck] === 0) {
+            ledBeatState = true;
+        }
+        if (PioneerDDJSX.slicerType[deck] === 1) {
+            ledBeatState = false;
+            if (((PioneerDDJSX.slicerBeatsPassed[deck] - 1) % 8) === 7 && !PioneerDDJSX.slicerAlreadyJumped[deck] && PioneerDDJSX.slicerPreviousBeatsPassed[deck] < PioneerDDJSX.slicerBeatsPassed[deck]) {
+                engine.setValue(group, "beatjump", -8);
```

Comment:
The slicer section size should be configurable, not hardcoded at 8. How about making it adjustable with shift + parameter left/right?